### PR TITLE
Add cache control support

### PR DESCRIPTION
**Pull request by richtier**

### CODE REVIEW

File: directory_client_core/cache_control.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import abc
+
+
+class AbstractCacheControl(abc.ABC):
```
Comment (richtier) [inline directory_client_core/cache_control.py]: allowing for flexibility of supporting in future `ModifiedSinceCacheControl` `NoCacheCacheControl`, et al

File: tests/test_base.py [hunk 2]
```diff
@@ -92,6 +92,28 @@ def test_send_response_not_ok(self, stub):
         response = self.client.send(method="POST", url="https://example.com")
         assert response.status_code == http.client.BAD_REQUEST
 
+    @stub_request('https://example.com/test', 'get')
```
Comment (kowalcj0) [inline tests/test_base.py]: @richtier can we have tests that verify whether client was served with cached or uncached response?

Comment (richtier) [reply tests/test_base.py]: [we do](https://github.com/uktrade/directory-cms-client/blob/master/tests/test_client.py#L194) :)

File: directory_client_core/base.py [hunk 3]
```diff
@@ -109,16 +110,19 @@ def build_url(base_url, partial_url):
...
         headers = {
             "User-agent": "EXPORT-DIRECTORY-API-CLIENT/{}".format(self.version)
         }
+
         if authenticator:
-            headers.update(authenticator.get_auth_headers())
+            headers.update(authenticator.headers)
+
+        if cache_control:
+            headers.update(cache_control.headers)
```
Comment (kowalcj0) [inline directory_client_core/base.py]: FYI: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/If-None-Match
```...Note that the server generating a 304 response MUST generate any of the following header fields that would have been sent in a 200 (OK) response to the same request: Cache-Control, Content-Location, Date, ETag, Expires, and Vary...```
are those aforementioned headers set here?

Comment (richtier) [reply directory_client_core/base.py]: yes, `ETag` is set :)

**Pull request merged**